draco/asp/hard.lp

```diff
@@ -230,6 +230,27 @@ violation(bar_area_without_zero) :-
     not attribute((scale,zero),S,_),
     C = (x;y).
 
+% @hard(row_or_col_c) Row and column require discrete. If number, need to be binned.
+violation(row_or_col_c) :-
```

Comment:
I think field with number type can be discrete too. For example if it's ordinal with a small cardinality.

Reply:
@domoritz Do we want to write this one in soft constraints considering numeric field can be ordinal(but can't be explicitly indicated)?

Reply:
Yes, a numeric field can be discrete as well. We could have soft constraints that require high cardinality numbers to be binned to be discrete.

Reply:
So we don't need this hard constraint to require numbers must be binned for faceting?

Reply:
I don't think we should have this as a hard constraint. The soft constraint should be a bit more thoughtful like "the cardinality of facets should be less than x" and then we should define the effective cardinality based on the data cardinality and binning. We also need to add a generate for binning. Then the silver will recommend binning when the cardinality is too high.